dev/Deployment/DeploymentManager.cpp

```diff
@@ -327,29 +337,38 @@ namespace winrt::Microsoft::Windows::ApplicationModel::WindowsAppRuntime::implem
     HRESULT DeploymentManager::VerifyPackage(const std::wstring& packageFamilyName, const PACKAGE_VERSION targetVersion,
         __out std::wstring& packageIdentifier) try
     {
-        auto packageFullNames{ FindPackagesByFamily(packageFamilyName) };
+        winrt::Windows::Management::Deployment::PackageManager packageManager;
+        auto packages{ packageManager.FindPackagesForUser(L"", packageFamilyName) };
+
         bool match{};
-        for (const auto& packageFullName : packageFullNames)
+        bool matchedPackageStatusIsOK{};
+        for (const auto& package : packages)
         {
-            auto packagePath{ GetPackagePath(packageFullName) };
+            auto packagePath{ package.InstalledPath()};
             if (packagePath.empty())
             {
                 continue;
             }
 
-            auto packageId{ AppModel::Identity::PackageIdentity::FromPackageFullName(packageFullName.c_str()) };
-            if (packageId.Version().Version >= targetVersion.Version)
+            UINT64 packageVersion = (static_cast<UINT64>(package.Id().Version().Major) << 48) +
+                                    (static_cast<UINT64>(package.Id().Version().Minor) << 32) +
+                                    (static_cast<UINT64>(package.Id().Version().Build) << 16) +
+                                    (static_cast<UINT64>(package.Id().Version().Revision));
```

Comment:
Simpler if you use `PACKAGE_VERSION` e.g.

```
const auto packageVersion{ ::AppModel::Package::ToPackageVersion(package.Id().Version()).Version };
```

and add this to `dev\common\AppModel.Package.h`

```
    inline PACKAGE_VERSION ToPackageVersion(winrt::Windows::ApplicationModel::PackageVersion const& from)
    {
        PACKAGE_VERSION to{};
        to.Major = from.Major;
        to.Minor = from.Minor;
        to.Build = from.Build;
        to.Revision = from.Revision;
        return to;
    }
```